### PR TITLE
Injector refactorings: moved pod patch to a separate library

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
# Description

This PR doesn't include any new feature. It is purely a refactoring of the injector, cleaning up the code that does the pod patching and moving that to a separate library.

This is part of the work for #4730

## Issue reference

Part of #4730

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [X] Unit tests passing
* [X] End-to-end tests passing

### CODE REVIEW

File: pkg/injector/annotations/annotations.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	DefaultHealthzProbePeriodSeconds    = 6
+	DefaultHealthzProbeThreshold        = 3
+	DefaultMtlsEnabled                  = true
+	DefaultEnableAPILogging             = false
+	DefaultEnableProfiling              = false
+	DefaultDisableBuiltinK8sSecretStore = false
+	DefaultEnableAppHealthCheck         = false
+	DefaultAppCheckPath                 = "/health"
+	DefaultAppHealthProbeInterval       = 5   // in seconds
+	DefaultAppHealthProbeTimeout        = 500 // in ms
```
Comment (shubham1172) [inline pkg/injector/annotations/annotations.go:90]: Shall we update these?
```suggestion
	DefaultAppHealthProbeIntervalSeconds            = 5
	DefaultAppHealthProbeTimeoutMilliSeconds        = 500
```

Comment (artursouza) [reply pkg/injector/annotations/annotations.go:90]: I agree.

Comment (ItalyPaleAle) [reply pkg/injector/annotations/annotations.go:90]: I will change that

Comment (ItalyPaleAle) [reply pkg/injector/annotations/annotations.go]: Done

File: pkg/injector/sidecar/pod-patch.go [hunk 2]
```diff
@@ -0,0 +1,224 @@
+/*
```
Comment (shubham1172) [inline pkg/injector/sidecar/pod-patch.go:1]: nit: filename of some new files are currently in the form of `X-Y.go`, should they rather be `X_Y.go` (or `XY.go`)? https://github.com/golang/go/issues/36060#issuecomment-563435344

File: pkg/injector/sidecar/sidecar-container.go [hunk 3]
```diff
@@ -0,0 +1,425 @@
+/*
```
Comment (shubham1172) [inline pkg/injector/sidecar/sidecar-container.go:1]: Filename suggestion: `container.go`

File: pkg/injector/sidecar/utilts_test.go [hunk 4]
```diff
@@ -0,0 +1,178 @@
+/*
```
Comment (shubham1172) [inline pkg/injector/sidecar/utilts_test.go:1]: typo in filename

**Review by shubham1172 — APPROVED** [review]

**Comment by ItalyPaleAle** [comment]
@shubham1172  thanks for the review! I made all the changes you requested except changing the method name to `getContainer` because I think being explicit on what container it is may be good :)

**Comment by shubham1172** [comment]
> @shubham1172 thanks for the review! I made all the changes you requested except changing the method name to `getContainer` because I think being explicit on what container it is may be good :)

Makes sense, thanks!

**Review by artursouza — APPROVED** [review]

**Pull request merged**